### ISSUE TITLE
Missing Links: Some individuals are incorrectly showing that they have no parents or children.

### ISSUE THREAD

**Issue by jmegenealogy** [wikitree/wikitree-dynamic-tree#280]
Two separate issues:

1. Profiles with Unlisted privacy sometimes aren't listed as having parents even if they do

* https://www.wikitree.com/g2g/1816375/cc7-views-for-west-14805-degree-3

2. Profiles in the last degree don't show children even if they have them

This is probably because another degree is necessary to calculate the children. Update ML to automatically click the "additional degree" box when gathering data, if the URL parameter is used.

* https://www.wikitree.com/g2g/1815876/missing-links-is-now-part-of-cc7-views-in-tree-apps?show=1816162#a1816162

**Comment by udjeni** [comment]
As far as I am aware, you will not get the parents of an unlisted profile. Their id in the API data is negative and they are just not listed amongst the children of the parents. Unless something changed in the API, the best solution would be to not include them in your missing links calculation.

I would not automatically click the improve accuracy checkbox, but would leave it for the user to decide (because for some people, if they would say get cc7, you will just fail because their cc8 pushes them over the limit. My recommendation would be to just allow the accuracy checkbox on the missing links page. Then, if you want, you could also colour the max retrieved degree profiles' missing children (and possibly spouses?) with a different colour, because it will only be them that will have wrong counts. Parent counts should always be correct (except for some private profiles - the data being returned for private profiles are quite inconsistent).

**Comment by jmegenealogy** [comment]
> As far as I am aware, you will not get the parents of an unlisted profile. Their id in the API data is negative and they are just not listed amongst the children of the parents. Unless something changed in the API, the best solution would be to not include them in your missing links calculation.

If someone is on the TL of the Unlisted profile, the parents should be available to them. 

Actually, I think this might be another issue with not getting the next generation. When I just tested, my unlisted uncles show as no parents at CC3, but will have parents at CC4.

I'm only planning on automatically checking the box if the URL parameter for ML is used. Since the people are usually going to be coming from elsewhere, the goal is for it to just work for them where they don't have to mess with the settings.

**Comment by udjeni** [comment]
> my unlisted uncles show as no parents at CC3, but will have parents at CC4.

Did you check that? :)

> I'm only planning on automatically checking the box if the URL parameter for ML is used. Since the people are usually going to be coming from elsewhere, the goal is for it to just work for them where they don't have to mess with the settings

In order to be consistent with what the user will see on the Table view, then I think that you should make sure that the GET button is labelled the same (but that should probably happen automatically). Also, I think it would be good (for consistency) to have the Improve accuracy checkbox visible on the Missing Links page.

**Comment by jmegenealogy** [comment]
> Did you check that? :)

Yes (these are uncles by marriage, so the spouses of my parent's siblings, so their parents aren't in CC2).

**Comment by udjeni** [comment]
But did you see the parents linked to them when you retrieved next CC level up? The profiles of the parents might be present, but are the unlisted children being counted and do the unlisted children now show the correct number of parents?
